src/connectors/articleService.ts

```diff
@@ -334,14 +335,69 @@ export class ArticleService extends BaseService<Article> {
 
     const articleVersion = await this.models.create({
       table: 'article_version',
-      data: { ...data, ...newData } as Partial<ArticleVersion>,
+      data: { ...data, ...newData, description } as Partial<ArticleVersion>,
     })
+    this.latestArticleVersionLoader.clear(articleId)
     return articleVersion
   }
 
   public countArticleVersions = async (articleId: string) =>
     this.models.count({ table: 'article_version', where: { articleId } })
 
+  public findArticleVersions = async (
+    articleId: string,
+    { take, skip }: { take?: number; skip?: number } = {},
+    onlyContentChange = true
+  ) => {
+    const records = await this.knexRO('article_version')
+      .select('*', this.knexRO.raw('COUNT(1) OVER() ::int AS total_count'))
+      .from(
+        this.knexRO('article_version')
+          .where({ articleId })
+          .modify((builder) => {
+            if (onlyContentChange) {
+              builder.select(
+                '*',
+                this.knexRO.raw(
+                  'LAG(content_id, 1) OVER(order by id) AS last_content_id'
+                ),
+                this.knexRO.raw(
+                  'LAG(content_md_id, 1) OVER(order by id) AS last_content_md_id'
+                )
+              )
+            } else {
+              builder.select('*')
+            }
+          })
+          .orderBy('id', 'desc')
+          .as('t')
+      )
+      .modify((builder) => {
+        if (onlyContentChange) {
+          builder
+            .where('content_id', '!=', this.knexRO.ref('last_content_id'))
+            .orWhere(
+              'content_md_id',
+              '!=',
+              this.knexRO.ref('last_content_md_id')
+            )
+            .orWhere((whereBuilder) => {
+              // first version
+              whereBuilder
+                .where('last_content_id', null)
+                .where('last_content_md_id', null)
```

Comment:
these would convert to `= null` test, may not work as you may need a `IS NULL` test,

better use `.whereNull` ?
https://knexjs.org/guide/query-builder.html#wherenull